app/stv/stv.js

```diff
@@ -0,0 +1,360 @@
+      if (winners.length === seats) {
+        return {
+          result: { status: 'RESOLVED', winners },
+          log,
+          thr,
+        };
+      }
+
+      // Set the done alternatives as the roundwinners
+      doneAlternatives = roundWinners;
+
+      // The next rounds votes are votes that are not done.
+      nextRoundVotes = votes.filter((_, i) => !doneVotes[i]);
+
+      // Go through all done votes
+      for (let i in doneVotes) {
+        // @const { Vote } vote - The vote for this loop
+        const vote = JSON.parse(JSON.stringify(votes[i]));
+
+        // @const { Alternative } alternative - Take the first choice of the done vote
+        const alternative = JSON.parse(JSON.stringify(vote.priorities[0]));
+
+        // @const { float } fraction - Find the excess fraction for this alternative
+        const fraction = excessFractions[alternative._id] || 0;
+
+        // If the fraction is 0 (meaning no votes should be transferred) or if the vote
+        // has no more priorities (meaning it's exhausted) we can continue without transfer
+        if (fraction === 0 || vote.priorities.length === 1) continue;
+
+        // Fractional transfer. We mutate the weight for these votes by a fraction
+        vote['weight'] = vote.weight * fraction;
+        // Push the mutated votes to the list of votes to be processed in the next iteration
+        nextRoundVotes.push(vote);
+      }
+    } else {
+      // Find the lowest score
+      const minScore = Math.min(
+        ...alternatives.map(
+          (alternative) => counts[alternative.description] || 0
+        )
+      );
+
+      // Find the candidates with the lowest score
+      const minAlternatives = alternatives.filter(
+        (alternative) =>
+          (counts[alternative.description] || 0) <= minScore + EPSILON
+      );
+
+      // There is a tie for eliminating candidates. Per Scottish STV we must look at the previous rounds
+      if (minAlternatives.length > 1) {
+        let reverseIteration = iteration;
+        log.push({
+          action: 'TIE',
+          description: `There are ${
+            minAlternatives.length
+          } candidates with a score of ${Number(
+            minScore.toFixed(4)
+          )} at iteration ${reverseIteration}`,
+        });
+
+        // If the minScore is 0 or 1 we eliminate a random candidate as Scottish STV specifies
+        // This is only done in the low cases of 0 or 1, and we would rather return an unresolved
+        // election if the TIE cannot be solved by backtracking.
+        if (minScore === 0 || minScore === 1) {
+          const randomAlternative =
+            minAlternatives[Math.floor(Math.random() * minAlternatives.length)];
+          log.push({
+            action: 'RANDOMELIMINATE',
+            alternative: randomAlternative,
+            minScore: Number(minScore.toFixed(4)),
+          });
+          doneAlternatives[randomAlternative._id] = {};
```

Comment:
@JonasBak Made it so that for `minScore` of `0 || 1` we abide by the Scottish random rules. I think we can argue that this makes sense, as this would be a candidate with 0 votes, or someone that has just voted for themselves, and that this should not break an entire election.